### PR TITLE
Add PHPStan

### PR DESCRIPTION
**Pull request by swissspidy**
Working towards higher levels

### CODE REVIEW

File: build-cs/composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "phpstan/phpstan-phpunit": "^1.3",
+		"swissspidy/phpstan-no-private": "^v0.2.0",
+    "szepeviktor/phpstan-wordpress": "^1.3.0"
+  },
+  "config": {
+    "allow-plugins": {
+      "phpstan/extension-installer": true
+    }
+  }
+}
```
Comment (szepeviktor) [inline build-cs/composer.json:16]: TAB-s are mixed with spaces
```suggestion
{
  "require-dev": {
    "php-stubs/wordpress-tests-stubs": "^6.3.0",
    "phpstan/extension-installer": "^1.3.1",
    "phpstan/phpstan": "^1.10",
    "phpstan/phpstan-deprecation-rules": "^1.1",
    "phpstan/phpstan-phpunit": "^1.3",
    "swissspidy/phpstan-no-private": "^v0.2.0",
    "szepeviktor/phpstan-wordpress": "^1.3.0"
  },
  "config": {
    "allow-plugins": {
      "phpstan/extension-installer": true
    }
  }
}
```

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -54,6 +54,28 @@ jobs:
...
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: 'latest'
+                  coverage: none
+                  tools: composer, cs2pr
+
+            - name: Install PHP dependencies
+              uses: ramsey/composer-install@v2
+              with:
+                  composer-options: '--prefer-dist --no-progress --no-interaction'
```
Comment (szepeviktor) [inline .github/workflows/lint-and-test.yml:74]: These two are built in.
```suggestion
                  composer-options: '--prefer-dist'
```

File: .github/workflows/lint-and-test.yml [hunk 3]
```diff
@@ -54,6 +54,28 @@ jobs:
...
+
+            - name: Run PHPStan
+              run: composer phpstan
```
Comment (szepeviktor) [inline .github/workflows/lint-and-test.yml:77]: This is a script, not a command.
```suggestion
              run: composer run phpstan
```

File: .github/workflows/lint-and-test.yml (hunk 2 above)
Comment (swissspidy) [reply .github/workflows/lint-and-test.yml:74]: TIL! Thanks.

File: .github/workflows/lint-and-test.yml (hunk 3 above)
Comment (swissspidy) [reply .github/workflows/lint-and-test.yml:77]: It works though, but makes sense to be explicit 👍

**Comment by szepeviktor** [comment]
So sad to see low level code (HTTP request processing) mixed with high level code (language).

**Pull request merged**